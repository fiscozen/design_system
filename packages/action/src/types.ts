import { IconVariant } from "@fiscozen/icons";
import { FzLinkProps } from "@fiscozen/link";
import { RouteLocation } from "vue-router";

// Environment variants
export type FzActionEnvironment = "backoffice" | "frontoffice";

// Layout variants
export type FzActionVariant = "textLeft" | "textCenter" | "onlyIcon";

// Common props shared by both link and button actions
export interface FzActionCommonProps {
  /**
   * Name of the fontawesome icon
   */
  iconName?: string;
  /**
   * Variant of the fontawesome icon
   */
  iconVariant?: IconVariant;
  /**
   * Alternative prop to default label slot
   */
  label?: string;
  /**
   * Sub-label text (smaller text below the main label)
   */
  subLabel?: string;
  /**
   * Disables the action
   */
  disabled?: boolean;
  /**
   * Environment variant (backoffice/frontoffice)
   */
  environment?: FzActionEnvironment;
  /**
   * Layout variant (textLeft/textCenter/onlyIcon)
   */
  variant?: FzActionVariant;
  /**
   * Whether text should be truncated
   */
  isTextTruncated?: boolean;
}

// Props specific to link actions (from FzLink)

// Discriminated union for action props
export type FzActionProps =
  | (FzActionCommonProps &
      FzLinkProps & {
        type: "link";
        meta: CustomRouteLocation;
      })
  | (FzActionCommonProps & {
      type: "action";
    });

// Helper type for Vue Router route locations
type PartialExcept<T, K extends keyof T> = Pick<Required<T>, K> & Partial<T>;

export type CustomRouteLocation = PartialExcept<RouteLocation, "path" | "name">;
