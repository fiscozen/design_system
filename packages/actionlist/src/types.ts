import { FzRouterNavlinkProps, FzNavlinkProps } from '@fiscozen/navlink'

type ActionlistItem<T = void> =
  | (FzRouterNavlinkProps & { type: 'link' })
  | (FzNavlinkProps<T> & { type: 'button' })

interface FzActionlistProps<T = void> {
  label?: string;
  items: ActionlistItem<T>[];
  listClass?: string;
}

export { ActionlistItem, FzActionlistProps }
