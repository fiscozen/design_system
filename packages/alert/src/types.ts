import { RouteLocationRaw } from 'vue-router'

type AlertProps = {
  /**
   * Type of the topbar which dictates the appearance
   */
  type: 'info' | 'error' | 'danger' | 'warning' | 'success'
  /**
   * Style which dictates the action rendered
   */
  alertStyle?: 'default' | 'collapsable' | 'simple'
  /**
   * Whether to render the button action
   */
  showButtonAction?: boolean
  /**
   * Button action label
   */
  buttonActionLabel?: string
  /**
   * Button action tooltip
   */
  buttonActionTooltip?: string
  /**
   * Whether to render the link action
   */
  showLinkAction?: boolean
  /**
   * Link action label
   */
  linkActionLabel?: string
  /**
   * Action link location (applies only if showLinkAction is true)
   */
  linkActionLocation?: RouteLocationRaw
  /**
   * Size of the alert (applies only if style is simple)
   */
  size?: 'sm' | 'md' | 'lg'
  /**
   * Title of the alert (applies only if style is not simple)
   */
  title?: string
  /**
   * Default state of the alert (open/closed)
   */
  defaultOpen?: boolean
}

export { AlertProps }
