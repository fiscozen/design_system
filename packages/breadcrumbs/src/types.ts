import { RouteLocation, RouterLinkProps } from 'vue-router'

export interface Breadcrumb<T = void> {
  id: string
  label: string
  metadata: T extends {} ? T : undefined
}

type PartialExcept<T, K extends keyof T> = Pick<Required<T>, K> & Partial<T>

export type CustomRouteLocation = PartialExcept<RouteLocation, 'path' | 'name'>
