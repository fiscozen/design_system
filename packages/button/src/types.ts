/**
 * Type definitions for the Fiscozen Button component library.
 * 
 * @module @fiscozen/button/types
 */

import type { IconVariant, IconSize } from '@fiscozen/icons'

export type CommonButtonVariant = 'primary' | 'secondary' | 'invisible'
export type IconButtonVariant = CommonButtonVariant | 'notification'
export type ButtonVariant = CommonButtonVariant | 'danger' | 'success'
export type ButtonSize = 'xs' | 'sm' | 'md' | 'lg'
export type ButtonEnvironment = 'backoffice' | 'frontoffice'

/**
 * Type for size to environment mapping record
 */
export type SizeToEnvironmentMap = Record<ButtonSize, ButtonEnvironment>

/**
 * Type for button size to icon size mapping record
 */
export type IconSizeMap = Record<ButtonSize, IconSize>

/**
 * Props for the FzButton component.
 * 
 * A comprehensive button component supporting five visual variants and two environment-based sizes.
 * Features include icon integration, multiple interactive states, and full accessibility support.
 * 
 * @example
 * <FzButton label="Click me" variant="primary" />
 */
export interface FzButtonProps {
  /**
   * Text label displayed on the button. Overridden by default slot if provided.
   */
  label?: string
  /**
   * @deprecated Use FzTooltip component to wrap the button instead.
   * A runtime warning will be displayed when this prop is used.
   */
  tooltip?: string
  /**
   * Visual style variant determining colors and interactive states
   * @default 'primary'
   */
  variant?: ButtonVariant
  /**
   * Button size affecting height, padding, and text size
   * @deprecated Use the 'environment' prop instead. This prop will be removed in a future version.
   */
  size?: ButtonSize
  /**
   * Environment determining button size
   * @default 'frontoffice'
   */
  environment?: ButtonEnvironment
  /**
   * Disables interaction and applies disabled styling
   * @default false
   */
  disabled?: boolean
  /**
   * FontAwesome icon name (e.g., 'bell', 'save'). Search: https://fontawesome.com/v6/icons
   */
  iconName?: string
  /**
   * FontAwesome icon variant (e.g., 'fas', 'far', 'fal')
   */
  iconVariant?: IconVariant
  /**
   * Icon position relative to label
   * @default 'before'
   */
  iconPosition?: 'before' | 'after'
  /**
   * Additional CSS classes for label container. Merged with 'truncate' unless overrideContainerClass is true.
   */
  containerClass?: string
  /**
   * Replaces default container classes with containerClass instead of merging
   * @default false
   */
  overrideContainerClass?: boolean
}

/**
 * Props for the FzIconButton component.
 * 
 * Icon-only button variant with notification badge support.
 * 
 * @example
 * <FzIconButton iconName="bell" variant="notification" />
 */
export interface FzIconButtonProps {
  /**
   * The tooltip of the button will be shown on hover
   */
  tooltip?: string
  /**
   * primary or secondary button
   * @default 'primary'
   */
  variant?: IconButtonVariant
  /**
   * size of the button
   * @default 'md'
   */
  size?: ButtonSize
  /**
   * whether action is enabled or not
   * @default false
   */
  disabled?: boolean
  /**
   * Icon to be displayed. Use fontawesome search here https://fontawesome.com/v6/icons
   * Mind that not all of the icons and variants might be available.
   */
  iconName: string
  /**
   * Fontawesome icon variant: solid, regular, light, thin. Sharp subvariants are available as well
   * @default 'far'
   */
  iconVariant?: IconVariant
  /**
   * Icon size
   */
  iconSize?: IconSize
  /**
   * Positioning of the icon
   * @default 'before'
   */
  iconPosition?: 'before' | 'after'
  /**
   * Additional class for the icon
   * @default ''
   */
  iconClass?: string
}

/**
 * Props for the FzButtonGroup component.
 * 
 * Container component for grouping multiple buttons with consistent spacing.
 * 
 * @example
 * <FzButtonGroup horizontal :gap="true">
 *   <FzButton label="Save" />
 *   <FzButton label="Cancel" />
 * </FzButtonGroup>
 */
export interface FzButtonGroupProps {
  /**
   * The direction of the button group. If true, buttons will be displayed horizontally, otherwise vertically
   * @default true
   */
  horizontal?: boolean
  /**
   * Whether to add a gap between buttons
   */
  gap?: boolean
  /**
   * The gap size between buttons
   * @default 'md'
   */
  size?: ButtonSize
}
