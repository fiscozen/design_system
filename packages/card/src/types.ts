export type FzCardProps = {
  /**
   * The title of the card
   */
  title?: string;
  /**
   * The background color of the card
   */
  color?: FzCardColor;
  /**
   * The primary action button
   */
  primaryAction?: FzCardButton;
  /**
   * The secondary action button
   */
  secondaryAction?: FzCardButton;
  /**
   * The tertiary action button
   */
  tertiaryAction?: FzCardIconButton;
  /**
   * Custom css class to apply to the content
   */
  contentClass?: string;
  /**
   * Whether the card is collapsible
   */
  collapsible?: boolean;
  /**
   * Whether the card is expanded by default (only if collapsible)
   */
  defaultExpanded?: boolean;
  /**
   * Whether the card content is always alive (never destroyed) when collapsed
   */
  alwaysAlive?: boolean;
};

type FzCardButton = {
  label: string;
};

type FzCardIconButton = {
  icon: string;
};

export type FzCardColor = "purple" | "orange" | "blue";

export interface FzCardEvents {
  /**
   * Event emitted when the primary action is clicked
   * @type {() => void}
   */
  (event: "fzprimary:click"): void;
  /**
   * Event emitted when the secondary action is clicked
   * @type {() => void}
   */
  (event: "fzsecondary:click"): void;
  /**
   * Event emitted when the tertiary action is clicked
   * @type {() => void}
   */
  (event: "fztertiary:click"): void;
}

export interface FzCardSlots {
  /**
   * Slot for the content of the card
   */
  default(props: {}): any;
  /**
   * Slot for the header, it will be displayed on the left of the title
   */
  header(props: {}): any;
  /**
   * Slot for the header content, it will be displayed below the title
   */
  "header-content"(props: {}): any;
  /**
   * Slot for the footer of the card
   */
  footer(props: {}): any;
}
