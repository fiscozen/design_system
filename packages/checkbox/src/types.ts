import { FzTooltipProps } from "@fiscozen/tooltip";

export type FzCheckboxProps = {
  /**
   * The label of the checkbox
   */
  label: string;
  /**
   * The value of the checkbox. If not provided, the value will be the label
   */
  value?: string | number | boolean;
  /**
   * The size of the checkbox
   */
  size?: "sm" | "md";
  /**
   * if the checkbox is indeterminate
   */
  indeterminate?: boolean;
  /**
   * If true, the checkbox will be emphasized
   */
  emphasis?: boolean;
  /**
   * If true, the checkbox will be disabled
   */
  disabled?: boolean;
  /**
   * If true, the checkbox will be in an error state
   */
  error?: boolean;
  /**
   * If the checkbox is required
   */
  required?: boolean;
  /**
   * If the checkbox is standalone
   */
  standalone?: boolean;
  /**
   * Tooltip props for the checkbox
   */
  tooltip?: FzTooltipProps;
  /**
   * ARIA owns attribute for hierarchical checkboxes
   */
  ariaOwns?: string;
  /**
   * Custom ID for the checkbox (if not provided, a random ID will be generated)
   */
  checkboxId?: string;
};

export type FzCheckboxGroupProps = {
  /**
   * The label of the checkbox group
   */
  label: string;
  /**
   * The size of the checkbox
   */
  size: "sm" | "md";
  /**
   * The checkbox to render
   */
  options: ParentCheckbox[];
  /**
   * If true, the checkbox will be emphasized
   */
  emphasis?: boolean;
  /**
   * If true, the checkbox will be disabled
   */
  disabled?: boolean;
  /**
   * If true, the checkbox will be in an error state
   */
  error?: boolean;
  /**
   * If the checkbox group is required
   */
  required?: boolean;
  /**
   * If true, the checkbox group will be horizontal
   */
  horizontal?: boolean;
};
export type ParentCheckbox = ChildCheckbox & { children?: ChildCheckbox[] };
export type ChildCheckbox = Omit<FzCheckboxProps, "size">;
