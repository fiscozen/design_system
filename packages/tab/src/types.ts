export type FzTabsProps = {
  /**
   * Size variant
   */
  size: "sm" | "md";
  /**
   * Enable horizontal overflow
   */
  horizontalOverflow?: boolean;
  /**
   * Enable vertical direction
   */
  vertical?: boolean;
};

export type FzTabProps = {
  /**
   * Title of the tab
   */
  title: string;
  /**
   * Icon to display on the tab
   */
  icon?: string;
  /**
   * Badge text to display on the tab
   */
  badgeContent?: string;
  /**
   * Disable the tab
   */
  disabled?: boolean;
  /**
   * Show the tab content
   */
  initialSelected?: boolean;
};
