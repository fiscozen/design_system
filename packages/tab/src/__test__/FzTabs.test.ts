import { mount } from "@vue/test-utils";
import { describe, it, expect, beforeEach } from "vitest";
import { FzTab, FzTabs } from "..";
import { h } from "vue";
import { FzTabProps, FzTabsProps } from "../types";

const createWrapper = async (
  props: FzTabsProps,
  tab1Props: FzTabProps,
  tab2Props: FzTabProps,
) => {
  const content = mount(FzTabs, {
    props,
    slots: {
      default: () => [
        h(FzTab, tab1Props, "Content tab1"),
        h(FzTab, tab2Props, "Content tab2"),
      ],
    },
  });

  await content.vm.$nextTick();
  return content;
};

describe("FzTabs", () => {
  beforeEach(() => {
    globalThis.HTMLElement.prototype.scrollIntoView = () => {};
  });
  it("renders with base case", async () => {
    const wrapper = await createWrapper(
      {
        size: "sm",
      },
      {
        title: "tab1",
      },
      {
        title: "tab2",
      },
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders with md size", async () => {
    const wrapper = await createWrapper(
      { size: "md" },
      { title: "tab1" },
      { title: "tab2" },
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders with badgeContent on tab1", async () => {
    const wrapper = await createWrapper(
      { size: "sm" },
      { title: "tab1", badgeContent: "1" },
      { title: "tab2" },
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders with icon on tab1", async () => {
    const wrapper = await createWrapper(
      { size: "sm" },
      { title: "tab1", icon: "bell" },
      { title: "tab2" },
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("change tab", async () => {
    const wrapper = await createWrapper(
      { size: "sm" },
      { title: "tab1" },
      { title: "tab2" },
    );

    await wrapper.findAll("button[title='tab2']")[0].trigger("click");

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders with vertical direction", async () => {
    const wrapper = await createWrapper(
      { size: "sm", vertical: true },
      { title: "tab1" },
      { title: "tab2" },
    );

    expect(wrapper.html()).toMatchSnapshot();
  })
});
