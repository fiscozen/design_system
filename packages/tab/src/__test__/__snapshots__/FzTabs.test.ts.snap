// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzTabs > change tab 1`] = `
"<div data-v-97c498eb="" class="flex flex-col">
  <div data-v-97c498eb="" class="flex flex-row">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-row"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab1">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <!--v-if-->
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md bg-white text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>
  <!--v-if-->Content tab2
</div>"
`;

exports[`FzTabs > renders with badgeContent on tab1 1`] = `
"<div data-v-97c498eb="" class="flex flex-col">
  <div data-v-97c498eb="" class="flex flex-row">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-row"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md bg-white text-blue-500 cursor-pointer" title="tab1" badgecontent="1">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <div class="text-xs px-12 rounded-xl w-fit h-20 flex items-center font-medium bg-blue-500 text-core-white" size="sm">1</div>
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>Content tab1
  <!--v-if-->
</div>"
`;

exports[`FzTabs > renders with base case 1`] = `
"<div data-v-97c498eb="" class="flex flex-col">
  <div data-v-97c498eb="" class="flex flex-row">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-row"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md bg-white text-blue-500 cursor-pointer" title="tab1">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <!--v-if-->
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>Content tab1
  <!--v-if-->
</div>"
`;

exports[`FzTabs > renders with icon on tab1 1`] = `
"<div data-v-97c498eb="" class="flex flex-col">
  <div data-v-97c498eb="" class="flex flex-row">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-row"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md bg-white text-blue-500 cursor-pointer" title="tab1" icon="bell">
        <div class="flex items-center justify-center w-[15px] h-[15px]"><svg class="svg-inline--fa fa-bell fa-sm h-[12px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
            <path class="" fill="currentColor" d="M224 0c-17.7 0-32 14.3-32 32V51.2C119 66 64 130.6 64 208v25.4c0 45.4-15.5 89.5-43.8 124.9L5.3 377c-5.8 7.2-6.9 17.1-2.9 25.4S14.8 416 24 416H424c9.2 0 17.6-5.3 21.6-13.6s2.9-18.2-2.9-25.4l-14.9-18.6C399.5 322.9 384 278.8 384 233.4V208c0-77.4-55-142-128-156.8V32c0-17.7-14.3-32-32-32zm0 96c61.9 0 112 50.1 112 112v25.4c0 47.9 13.9 94.6 39.7 134.6H72.3C98.1 328 112 281.3 112 233.4V208c0-61.9 50.1-112 112-112zm64 352H224 160c0 17 6.7 33.3 18.7 45.3s28.3 18.7 45.3 18.7s33.3-6.7 45.3-18.7s18.7-28.3 18.7-45.3z"></path>
          </svg></div><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <!--v-if-->
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>Content tab1
  <!--v-if-->
</div>"
`;

exports[`FzTabs > renders with md size 1`] = `
"<div data-v-97c498eb="" class="flex flex-col">
  <div data-v-97c498eb="" class="flex flex-row">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-row"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-md h-40 gap-8 py-12 px-14 rounded-lg bg-white text-blue-500 cursor-pointer" title="tab1">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <!--v-if-->
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-md h-40 gap-8 py-12 px-14 rounded-lg text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>Content tab1
  <!--v-if-->
</div>"
`;

exports[`FzTabs > renders with vertical direction 1`] = `
"<div data-v-97c498eb="" class="flex flex-row">
  <div data-v-97c498eb="" class="flex flex-col">
    <div data-v-97c498eb="" class="tab-container flex rounded-lg gap-8 p-2 bg-grey-100 w-fit max-w-full overflow-x-auto flex-col"><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md bg-white text-blue-500 cursor-pointer" title="tab1">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab1</span>
        <!--v-if-->
      </button><button data-v-97c498eb="" class="w-auto flex font-medium items-center max-w-[136px] text-sm h-40 gap-6 py-8 px-12 rounded-md text-grey-500 bg-grey-100 hover:bg-background-alice-blue active:bg-white active:text-blue-500 cursor-pointer" title="tab2">
        <!--v-if--><span class="text-ellipsis whitespace-nowrap overflow-hidden">tab2</span>
        <!--v-if-->
      </button></div>
  </div>Content tab1
  <!--v-if-->
</div>"
`;
